src/app/login/[[...login]]/page.tsx

```diff
@@ -1,9 +1,95 @@
-import { SignIn } from "@clerk/nextjs";
+"use client";
+import { signIn } from "next-auth/react";
+import { Card, CardContent } from "@/components/ui/card";
+import { Button } from "@/components/ui/buttons/button";
+import { Lock } from "lucide-react";
+import logo from "@public/logos/xplorer_logo.svg";
+import Image from "next/image";
+import Loading from "@/app/loading";
+import { useAppContext } from "@/context/AppContext";
+import { useState } from "react";
 
 export default function Page() {
+  const { isLoading: isSessionLoading } = useAppContext();
+  const [isSigningIn, setIsSigningIn] = useState(false);
+
+  const handleSignIn = async () => {
+    setIsSigningIn(true);
+    try {
+      await signIn("okta", { callbackUrl: "/" });
+    } catch (error) {
+      console.error("Sign in error:", error);
+      setIsSigningIn(false);
+    }
+  };
+
```

Comment:
1.`  if (isSigningIn) return;` prevents multiple clicks on the sign-in button while an authentication request is in progress.
2. `useCallback` prevents unnecessary re-creations on re-renders.
3. Use `.finally()` instead of setting `isSigningIn` in both `try` and `catch`.

```suggestion
const handleSignIn = useCallback(async () => {
  if (isSigningIn) return;
  setIsSigningIn(true);

  try {
    await signIn("okta", { callbackUrl: "/" });
  } catch (error) {
    console.error("Sign in error:", error);
  } finally {
    setIsSigningIn(false);
  }
}, [isSigningIn]);
```

Reply:
thanks @20was da, have added this now